### PR TITLE
feat: Enable cli e2e tests in CI

### PR DESCRIPTION
**Pull request by ltyu**
### Description
This PR enables cli e2e ts tests and adds some tests for Warp apply

### Related issues
https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/4388

### Backward compatibility
Yes

### CODE REVIEW

File: typescript/cli/package.json [hunk 1]
```diff
@@ -44,7 +44,8 @@
     "clean": "rm -rf ./dist",
     "lint": "eslint . --ext .ts",
     "prettier": "prettier --write ./src ./examples",
-    "test": "mocha --config .mocharc.json",
+    "test": "yarn test:e2e",
+    "test:e2e": "./scripts/all-test.sh",
```
Comment (jmrossy) [inline typescript/cli/package.json:48]: Why separate these two?

File: typescript/cli/tmp/anvil2/warp-route-deployment-anvil2.yaml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+anvil2:
```
Comment (jmrossy) [inline typescript/cli/tmp/anvil2/warp-route-deployment-anvil2.yaml:1]: Did you mean to include these? If so, let's not put them in `cli/tmp`, the `tmp` makes it seem like they shouldn't be checked in

Comment (ltyu) [reply typescript/cli/tmp/anvil2/warp-route-deployment-anvil2.yaml]: sorry this was meant to be a draft to debug. Unfortunately i could not get this working with CI yet. The test hangs and then eventually fails (after 5 hours). 

I've concluded that the e2e tests do fails. My hunch is that it's anvil related, possibly conflicting with the foundry test script in the SDK. 

Will need more time to investigate the raw logs.

Comment (ltyu) [reply typescript/cli/tmp/anvil2/warp-route-deployment-anvil2.yaml]: Will remove

File: typescript/cli/package.json (hunk 1 above)
Comment (ltyu) [reply typescript/cli/package.json]: Fixed

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -114,7 +114,7 @@ jobs:
           fi
 
   yarn-test:
-    runs-on: ubuntu-latest
+    runs-on: larger-runner
```
Comment (paulbalaji) [inline .github/workflows/test.yml:117]: oh dang, how come?

would it make sense to put the e2e on a separate job?

Comment (jmrossy) [reply .github/workflows/test.yml:117]: +1, these cost more than $1 each run. We want to run only the bare minimum on them

File: typescript/cli/src/tests/warp-apply.e2e-test.ts [hunk 4]
```diff
@@ -35,7 +34,7 @@ const WARP_CORE_CONFIG_PATH_2 = `${REGISTRY_PATH}/deployments/warp_routes/ETH/an
 
 describe('WarpApply e2e tests', async function () {
   let chain2Addresses: ChainAddresses = {};
-  this.timeout(0); // No limit timeout since these tests can take a while
+  this.timeout(60000); // Long timeout since these tests can take a while
```
Comment (jmrossy) [inline typescript/cli/src/tests/warp-apply.e2e-test.ts:37]: I would have expected even longer than one minute actually
Btw, you can use underscores in numbers these days for legibility so `60000` -> `60_000`

File: .github/workflows/test.yml (hunk 3 above)
Comment (ltyu) [reply .github/workflows/test.yml:117]: Good to know! i'll try this without the larger-runner

File: typescript/cli/src/tests/warp-apply.e2e-test.ts (hunk 4 above)
Comment (ltyu) [reply typescript/cli/src/tests/warp-apply.e2e-test.ts]: Fixed

**Review by paulbalaji — APPROVED** [review]

**Review by jmrossy — APPROVED** [review]

**Pull request merged**